src/lib.rs

```diff
@@ -1,3 +1,75 @@
+//! Landlock is a security feature available since Linux 5.13.
+//! The goal is to enable to restrict ambient rights
+//! (e.g., global filesystem access)
+//! for a set of processes by creating safe security sandboxes as new security layers
+//! in addition to the existing system-wide access-controls.
+//! This kind of sandbox is expected to help mitigate the security impact of bugs,
+//! unexpected or malicious behaviors in applications.
+//! Landlock empowers any process, including unprivileged ones, to securely restrict themselves.
+//! More information about Landlock can be found in the [official website](https://landlock.io).
+//!
+//! This crate provides a safe abstraction for the Landlock system calls, along with some helpers.
+//!
+//! # Use cases
+//!
+//! This crate is especially useful for build-in application sandboxing:
+//! * Parsers hardening (e.g., archive tools, file format conversion, renderers).
+//! * (Part of) applications with limited file renaming or linking needs
+//!   (e.g., some system or network services).
+//! * Monolith applications dealing with different levels of confidentiality
+//!   (e.g., web browser, email server).
+//!
+//! # Examples
+//!
+//! Examples can be found with the [`Ruleset` documentation](Ruleset)
+//! and the [sandboxer example](https://github.com/landlock-lsm/rust-landlock/blob/master/examples/sandboxer.rs).
+//!
+//! # Current limitations
+//!
+//! This crate handles the Landlock features available since Linux 5.13
+//! and then inherits some [kernel limitations](https://www.kernel.org/doc/html/latest/userspace-api/landlock.html#current-limitations)
+//! that will be addressed with future kernel releases:
+//! * File reparenting: renaming or linking a file to a different parent directory is always denied.
+//! * Filesystem topology modification: arbitrary mounts are always denied.
+//!
+//! # Compatibility
+//!
+//! This crate is designed to enable security as much as possible on supported systems.
+//! Compatibility should only be about the system supporting Landlock,
+//! not about the required action dependencies (e.g., opening a file to add it to a rule).
+//! This enables to check that the use of Landlock would be OK if supported,
+//! instead of ignoring files that should be used in a Landlock ruleset.
+//! When running the same program on different environments,
+//! this helps catch issues with the assumed context
+//! (e.g., whether a directory hierarchy exists or not).
+//!
+//! To make it easier to migrate to a new version of this library,
+//! we use the builder pattern
+//! and designed objects to require the minimal set of method arguments.
+//! Most `enum` are marked as `non_exhaustive` to enable backward-compatible evolutions.
+//!
+//! A simple approach would have been to expose the Landlock [ABI] version
```

Comment:
I would suggest to phrase this more in terms of "what users are supposed to do", for example:

> In best-effort mode, users of the crate may identify [with the XYZ helper?] which Landlock features are deemed required and which features may be downgraded to use lower security on systems where they can't be enforced. It is discouraged to compare the system's provided Landlock ABI version directly, as it is difficult to track detailed ABI differences, and these will already be handled by the crate's "best effort" mechanism.

NB, I do not remember exactly right now how the difference between 'required' and 'best effort' is expressed in the Rust library for individual features - In my memory this was just a single global toggle?

Reply:
I changed with this:
```
//! With [`set_best_effort()`](Compatible::set_best_effort),
//! users of the crate may identify which Landlock features are deemed required
//! and which features may be downgraded to use lower security on systems
//! where they can't be enforced.
//! It is discouraged to compare the system's provided [Landlock ABI](ABI) version directly,
//! as it is difficult to track detailed ABI differences
//! which are handled thanks to the [`Compatible`] trait.
```